src/components/step-flow/step-flow.test.tsx

```diff
@@ -46,6 +46,60 @@ function calculateStepStateIndexes(
   return [stepsBefore, stepsAfter];
 }
 
+test("when aria-label is specified, the attribute is applied to the expected element", () => {
+  render(
+    <StepFlow
+      data-role="step-flow"
+      aria-label="foo"
+      title="foo"
+      currentStep={5}
+      totalSteps={6}
+      category="bar"
+      ref={() => {}}
+    />,
+  );
+
+  const stepFlowComponent = screen.getByTestId("step-flow");
+
+  expect(stepFlowComponent).toHaveAttribute("aria-label", "foo");
```

Comment:
**suggestion (non-blocking):** We could use the [`toHaveAccessbleName`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohaveaccessiblename) matcher here to ensure the accessible name has been correctly set. 

```suggestion
  expect(stepFlowComponent).toHaveAccessibleName( "foo");
```